crates/swc/tests/fixture/issue-3609/1/output/index.js

```diff
@@ -0,0 +1,25 @@
+function _defineProperty(obj, key, value) {
+    if (key in obj) {
+        Object.defineProperty(obj, key, {
+            value: value,
+            enumerable: true,
+            configurable: true,
+            writable: true
+        });
+    } else {
+        obj[key] = value;
+    }
+    return obj;
+}
+var _ref = `lit${"lit"}`, _ref1 = `tpl${`tpl`}`, _ref2 = `lit${1 + 1}`, _ref3 = `complex${"123".length()}`;
+class Foo {
+    constructor(){
+        _defineProperty(this, 1, 2);
+        _defineProperty(this, "3", 4);
+        _defineProperty(this, `5`, 6);
+        _defineProperty(this, _ref, "lit");
+        _defineProperty(this, _ref1, `tpl`);
+        _defineProperty(this, _ref2, `lit`);
+        _defineProperty(this, _ref3, `complex`);
```

Comment:
Babel output

```javascript
_ref = `complex${"123".length()}`;

class Foo {
  constructor() {
    _defineProperty(this, 1, 2);

    _defineProperty(this, "3", 4);

    _defineProperty(this, `5`, 6);

    _defineProperty(this, `lit${"lit"}`, "lit");

    _defineProperty(this, `tpl${`tpl`}`, `tpl`);

    _defineProperty(this, `lit${1 + 1}`, `lit`);

    _defineProperty(this, _ref, `complex`);
  }
}
```

Reply:
Is it ok to leave it like this?
Now Babel goes further by treating `lit${1+1}` as literal as well.

Reply:
Which do you prefer? Go with this + another PR, or improve with this PR?

Reply:
I need to determine the scope of the impact when changing `is_literal`.
I prefer another PR.